allennlp/training/trainer.py

```diff
@@ -673,6 +676,23 @@ def train(self) -> None:
                 formatted_time = time.strftime("%H:%M:%S", time.gmtime(estimated_time_remaining))
                 logger.info("Estimated training time remaining: %s", formatted_time)
 
+            epochs_trained += 1
+
+        training_elapsed_time = time.time() - training_start_time
+        metrics = {
+                "training_duration": time.strftime("%H:%M:%S", time.gmtime(training_elapsed_time)),
+                "training_start_epoch": epoch_counter,
+                "training_epochs": epochs_trained
+        }
+        if train_metrics:
```

Comment:
see above comment, if you use an empty dict to represent no values then you don't have to check this